notebooks/config.ipynb

```diff
@@ -0,0 +1,1046 @@
+      "    :param stdin_ok: OK to read from ``stdin``?\n",
+      "    :return: An initialized ``YAMLConfig`` object.\n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "help(config.get_yaml_config)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "606da5b3-4bff-4148-a9a5-908aa7dd5e8c",
+   "metadata": {},
+   "source": [
+    "`get_yaml_config()` can take input from a Python `dict` or a YAML file like the one below.\n",
+    "<!--cell 4-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 3,
+   "id": "c6e049df-38f6-4879-8e0d-68356226d94b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting: Hello\n",
+      "recipient: World\n"
+     ]
+    }
+   ],
+   "source": [
+    "%%bash\n",
+    "cat fixtures/config/get-config.yaml"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c2f72448-d35e-4a0b-a371-cb47c7b3338b",
+   "metadata": {},
+   "source": [
+    "Paths to config files can be provided either as a string or <a href=\"https://docs.python.org/3/library/pathlib.html#pathlib.Path\">Path</a> object. Since `get_yaml_config()` is used here, a `YAMLConfig` object is returned.\n",
+    "<!--cell 6-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 4,
+   "id": "cc3020a6-4eb4-4830-9263-a9fc8fac7450",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "greeting: Hello\n",
+      "recipient: World\n"
+     ]
+    }
+   ],
+   "source": [
+    "config1 = config.get_yaml_config(\n",
+    "    config=Path(\"fixtures/config/get-config.yaml\")\n",
+    ")\n",
+    "print(config1)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b7bcd736-ff78-4e8b-957f-b348b812c5f6",
+   "metadata": {},
+   "source": [
+    "Providing a Python `dict` will create a UW `Config` object with format matching the function used.\n",
+    "<!--cell 8-->"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 5,
+   "id": "f01ac223-4a02-40ba-822f-8e66ad39f313",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "message:\n",
+      "  greeting: Hi\n",
+      "  recipient: Earth\n"
+     ]
+    }
+   ],
+   "source": [
+    "input_config = {\"message\": {\"greeting\":\"Hi\", \"recipient\":\"Earth\"}}\n",
+    "config2 = config.get_yaml_config(\n",
+    "    config=input_config\n",
+    ")\n",
+    "print(config2)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "dc745e95-d1ce-435c-a488-13b761979e36",
+   "metadata": {},
+   "source": [
+    "## Config Depth Limitations\n",
+    "\n",
+    "Some config formats have limitations on the depth of their nested configs. Shell configs, for example, contain key-value pairs using the standard bash syntax `key=value` and can only contain key-value pairs at the top level.\n",
```

Comment:
```suggestion
    "Some config formats have limitations on the depth of their nested configs. Shell configs, for example, may only contain top-level, bash-syntax `key=value` pairs.\n",
```
Hopefully still clear, and shorter -- avoids repetition of `key-value pairs` and then the more-or-less-same `key=value` term. I don't think we have to say _too_ much, as the meaning is made quite clear by the example config just below.